CarePartner/Views/AccountSettingsView.swift

```diff
@@ -6,66 +6,196 @@
+            Image(decorative: "Tidepool Logo")
+                .resizable()
+                .aspectRatio(contentMode: .fill)
+                .frame(width: 150, height: 150)
+                .onLongPressGesture(minimumDuration: 2) {
+                    if !client.hasSession {
+                        UINotificationFeedbackGenerator().notificationOccurred(.warning)
+                        isEnvironmentActionSheetPresented = true
+                    }
                 }
+                .actionSheet(isPresented: $isEnvironmentActionSheetPresented) { environmentActionSheet }
+        }
+    }
+
+    private var environmentActionSheet: ActionSheet {
+        var buttons: [ActionSheet.Button] = environments.map { environment in
+            .default(Text(environment.description)) {
+                error = nil
+                selectedEnvironment = environment
             }
         }
+        buttons.append(.cancel())
+
+
+        return ActionSheet(title: Text(NSLocalizedString("Environment", comment: "Tidepool login environment action sheet title")),
+                           message: Text(selectedEnvironment.description), buttons: buttons)
     }
 
-    private var dismissButton: some View {
-        Button(action: { dismiss() } ) {
-            Text("Done").bold()
+    private var loginButton: some View {
+        Button(action: {
+            loginButtonTapped()
+        }) {
+            if isLoggingIn {
+                ProgressView()
+                    .progressViewStyle(CircularProgressViewStyle())
+            } else {
+                Text(NSLocalizedString("Login", comment: "Tidepool login button title"))
+            }
         }
-    }}
+        .buttonStyle(ActionButtonStyle())
+        .disabled(isLoggingIn)
+    }
+
+
+    private var logoutButton: some View {
+        Button(action: {
+            showingDeletionConfirmation = true
+        }) {
+            Text(NSLocalizedString("Logout", comment: "Logout button title"))
+        }
+        .buttonStyle(ActionButtonStyle(.secondary))
+        .disabled(isLoggingIn)
+    }
+
+    private func loginButtonTapped() {
+        guard !isLoggingIn else {
+            return
+        }
+
+        error = nil
+        isLoggingIn = true
+
+        Task {
+            do {
+                try await client.login(environment: selectedEnvironment, sceneDelegate: sceneDelegate)
+                isLoggingIn = false
+            } catch {
+                self.error = error
+                isLoggingIn = false
+            }
+        }
+    }
+
+    private var closeButton: some View {
+        Button(action: {
+            dismiss()
+        }) {
+            Text(closeButtonTitle)
+                .fontWeight(.regular)
+        }
+    }
+
+    private var closeButtonTitle: String { NSLocalizedString("Close", comment: "Close navigation button title of an onboarding section page view") }
+}
 
 struct SettingsView_Previews: PreviewProvider {
+    @MainActor
```

Comment:
Were you running into issues without this? You should never **_need_** to declare @MainActor at the view level.

Reply:
This can be removed.